### ISSUE TITLE
Skeleton for contributions

### ISSUE THREAD

**Issue by Rmano** [ra3xdh/qucs_s_docs#3]
Hi! 

What about creating a skeleton for the documentation for the QUCS-S manual? I have a bit of experience (I am the main maintainer of https://github.com/circuitikz/circuitikz).  I suggest that there could be a main document, in LaTeX report or book format, where the needed packages and the ad-hoc macros will be defined (maybe with a local subpackage, look at what I did [here](https://github.com/circuitikz/circuitikz/tree/master/doc).

This documentation will be more complex than my package, so probably a tree structure could be like: 
```
> tree
.
├── complexthing
│   └── complexthing.tex
├── imgs
├── introduction
│   ├── imgs
│   └── introduction.tex
├── macros.sty
├── manual-en.tex
├── tutorial
│   ├── imgs
│   └── tutorial.tex
└── whatever
```

and the main file could be

```latex
\documentclass{book}
\usepackage[T1]{fontenc}
\title{Title}
\author{Authors}
\begin{document}
\maketitle

Copyright and license notice.

\tableofcontents

%
\include{introduction/introduction.tex}
%
\include{tutorial/tutorial.tex}
%
%
\appendix
\include{complexthing/complexthing.tex}

```

This is a structure that allows to work in parallel on several part of the project (especially if, for example, the chapter are further subdivided in the structure), and compile incrementally (with `\includeonly`) just the part somebody is working on. We could add a GitHub action to compile the complete thing for each merge to `main` so that we can be sure to have the documentation always at least formally correct.

Several languages can be put in parallel --- for example, a `manual-ru.tex`  that could access the same `imgs/` subdirectories (the same for `tutorial/tutorial-ru.tex` for example). Not assigning numbers and letting LaTeX do its magic is important too, to add things flawlessly in the middle. 

What do you think? 

PD: another important decision is the LaTeX compiler to use. `pdflatex` or `lualatex`, basically. The first one is more stable, the second one is more powerful especially for multilanguage documents.

Another important thing is to decide how to organize the tutorial with respect to the fact that `qucs-s` can be used with several simulators. Probably assuming the default one is `ngspice`, and adding sections or subsections for the other simulation engines could be the easiest way to organize it. A special symbol in the margin could help separate the thing when it is needed.

**Comment by ra3xdh** [comment]
> so probably a tree structure could be like:

Yes, the proposed structure is acceptable. The sections could be converted to includes and placed in external files instead of one huge TeX file. You may open a PR with the proposed structure. 

> another important decision is the LaTeX compiler to use. pdflatex or lualatex, basically.

The `pdflatex` suits all need for Qucs-S documentation and could be selected as the build tool.

> We could add a GitHub action to compile the complete thing for each merge to main

Yes, this could be great option. This could be integrated with main Qucs-S package gihub workflow to download compiled PDF and include it in the binary package. But the TeX sources should not be included in Qucs-S tarball to avoid compile-time latex dependency.

> Another important thing is to decide how to organize the tutorial with respect to the fact that qucs-s can be used with several simulators. 

The simplest option is to imply Ngspice everywhere if it is not specified explicitly. The Qucsator is mentioned only in the chapter 9.3 (RF simulation with Qucsator) of the existing tutorial.

**Comment by Rmano** [comment]
Ok, I'll try to migrate the existing documentation to this format one of these days --- as soon as RL gives me a bit of time --- and prepare a PR. Let's wait a bit and see if there are more suggestions by contributors...
